inc/pageutils.php

```diff
@@ -269,6 +269,8 @@ function sectionID($title,&$check) {
  * @return bool exists?
  */
 function page_exists($id, $rev = '', $clean = true, $date_at = false) {
+    $id = (explode('#', $id, 2))[0]; // #3608
+
```

Comment:
I cannot test myself at the moment, but could explode() return something that is an array without entry ‘0’?
(For $id such as null, false, etc?)